src/Kernel/Process.class.st

```diff
@@ -604,6 +604,31 @@ Process >> signalException: anException [
 	oldList ifNotNil: [self resume]
 ]
 
+{ #category : 'testing' }
+Process >> state [
+
+	self == Processor activeProcess ifTrue: [ ^ #active ].
+	"A process is #terminated if one of the following conditions is met:
+	(1) the receiver is a defunct process (suspendedContext = nil or pc = nil)
+	(2) the receiver is suspended in the endProcess method (we cannot use pragmas...
+	pragmas require the compiler, it generates an evil dependency.
+	It also is not available in minimal images)"
+	(suspendedContext isNil or: [
+		 suspendedContext isDead or: [
+			 suspendedContext method == (self class >> #endProcess) ] ])
+		ifTrue: [ ^ #terminated ].
```

Comment:
Three conditions for a terminated state is definitely a place to improve the termination. But it is for another step.

For example there are several places where suspendedContext is set to nil in the middle of debugger operations (see  suspendedContext writers). Any stepping over these methods eventually will corrupt the process state.

Reply:
> But it is for another step.

glad to agree :)

Reply:
See my comment below—this is, indeed, a whoooole other conversation. I would be delighted if one of you wanted to take the initiative to make this into its own issue—which entails first figuring out what, exactly, that issue should _be_.